pkg/registry/memory/nse_server.go

```diff
@@ -34,7 +36,7 @@ import (
 type networkServiceEndpointRegistryServer struct {
 	networkServiceEndpoints NetworkServiceEndpointSyncMap
 	executor                serialize.Executor
-	eventChannels           []chan *registry.NetworkServiceEndpoint
+	eventChannels           map[string]chan *registry.NetworkServiceEndpoint
```

Comment:
How is this being protected against concurrency issues?

Reply:
This map is using only with executor.

Reply:
And I apparently was tired when I reviewed this... because I totally didn't see the executor just above it :(

Reply:
That's ok, sometimes better to ask than look :)